tests/e2e/resilience_test.go

```diff
@@ -234,6 +239,141 @@ func (tc *OperatorResilienceTestCtx) ValidateComponentsDeploymentFailure(t *test
 	tc.deleteZeroPodQuotaForOperator()
 }
 
+func (tc *OperatorResilienceTestCtx) ValidateMissingComponentsCRDHandling(t *testing.T) {
+	t.Helper()
+
+	// Get all CRDs
+	crds := tc.FetchResources(
+		WithMinimalObject(gvk.CustomResourceDefinition, types.NamespacedName{}),
+	)
+
+	// Find first occurrence of a CRD with ".components.platform.opendatahub.io" in its name
+	componentCrd := unstructured.Unstructured{}
+	for _, crd := range crds {
+		if strings.Contains(strings.ToLower(crd.GetName()), ".components.platform.opendatahub.io") {
+			componentCrd = crd
+			break
+		}
+	}
+
+	// Save a backup copy of the CRD
+	crdBackup := tc.resourceBackup(t, &componentCrd)
+	t.Cleanup(func() {
+		tc.EventuallyResourceCreatedOrUpdated(
+			WithObjectToCreate(crdBackup),
+			WithCustomErrorMsg("Failed to restore CRD from backup (cleanup)"),
+		)
+	})
+
+	// Delete the CRD to simulate missing CRD
+	propagationPolicy := metav1.DeletePropagationBackground
+	tc.DeleteResource(
+		WithMinimalObject(gvk.CustomResourceDefinition, types.NamespacedName{Name: componentCrd.GetName()}),
+		WithClientDeleteOptions(
+			&client.DeleteOptions{
+				PropagationPolicy: &propagationPolicy,
+			}),
+		WithWaitForDeletion(true),
+	)
+
+	// Validate pod health and system health
+	selector := tc.getOperatorPodSelector()
+	tc.validatePodHealth(t, selector)
+
+	// Manually restore the CRD from backup
+	tc.EventuallyResourceCreatedOrUpdated(
+		WithObjectToCreate(crdBackup),
+		WithCustomErrorMsg("Failed to restore CRD from backup"),
+	)
```

Comment:
It’s good to restore this at the end, but you’re doing it twice; once here and again inside the `Cleanup`.